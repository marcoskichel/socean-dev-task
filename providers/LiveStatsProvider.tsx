import { createContext, useContext, useMemo } from 'react';

export interface LiveStats {
  title: string;
  value: string;
  hint: string;
  progress?: number;
}

interface LiveStatsContextProps {
  stats: LiveStats[];
}

export const LiveStatsContext = createContext<LiveStatsContextProps>({
  stats: [],
});

interface LiveStatsProviderProps {
  children: React.ReactNode;
}

const LiveStatsProvider = (props: LiveStatsProviderProps) => {
  const { children } = props;

  const value = useMemo<LiveStatsContextProps>(
    () => ({
      stats: [
        {
          title: 'Total Value Locked',
          value: '$61.70M USD',
          hint: 'SOL/USD: $87.9800',
        },
        {
          title: 'SOL/scnSOL in Pool',
          value: '701.31K SOL',
          hint: '684,440 scnSOL',
        },
        {
          title: '7D APY',
          value: '6.08%',
          hint: '12H APY: 7.14%',
        },
        {
          title: 'SOL/scnSOL Price',
          value: '1.0247 SOL',
          hint: '= 1 scnSOL',
        },
        {
          title: 'EPOCH 272',
          value: '19%',
          hint: 'ETA: 2d 5h 16m 10s',
          progress: 19,
        },
      ],
    }),
    [],
  );

  return (
    <LiveStatsContext.Provider value={value}>
      {children}
    </LiveStatsContext.Provider>
  );
};

export const useLiveStats = () => useContext(LiveStatsContext);

export default LiveStatsProvider;
