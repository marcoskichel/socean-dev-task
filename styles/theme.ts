import { createTheme, PaletteOptions, Theme } from '@mui/material';

export type CustomPallete = PaletteOptions & { bg: string };

const theme: Theme = createTheme({
  typography: {
    fontFamily: '"Manrope", "Helvetica", "Arial", sans-serif',
    h1: {
      fontWeight: 700,
      fontSize: '2.5rem',
      marginTop: '6rem',
      marginBottom: '1rem',
    },
    h6: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.5rem',
      letter: '-1%',
    },
    body1: {
      fontWeight: 400,
      fontSize: '0.875rem',
      lineHeight: '1rem',
    },
    button: {
      textTransform: 'none',
    },
  },
  components: {
    MuiCard: {
      defaultProps: {
        sx: {
          borderRadius: '1.5rem',
        },
      },
    },
  },
  palette: {
    mode: 'dark',
    background: {
      paper: '#232323',
    },
    primary: {
      main: '#FFFFFF',
    },
    secondary: {
      main: '#F24F83',
    },
    text: {
      primary: '#FFFFFF',
      secondary: '#A7A7A7',
    },
    action: {
      disabled: '#5C5C5C',
    },
  },
});

export default theme;
