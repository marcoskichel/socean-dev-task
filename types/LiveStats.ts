export interface LiveStats {
  title: string;
  value: string;
  hint: string;
  progress?: number;
}
