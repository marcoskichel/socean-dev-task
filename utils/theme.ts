import { createTheme, Theme } from "@mui/material";
import { red } from "@mui/material/colors";

const theme: Theme = createTheme({
  palette: {
    primary: {
      main: '#FFFFFF',
    },
    secondary: {
      main: '#A7A7A7',
    },
    action: {
      disabled: '#5C5C5C'
    }
  },
});

export default theme;