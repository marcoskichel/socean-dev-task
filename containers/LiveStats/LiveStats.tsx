import { styled } from '@mui/material';
import StatsItem from '../../components/StatsItem/StatsItem';
import { useLiveStats } from '../../providers/LiveStats/LiveStatsProvider';

const Root = styled('div')(({ theme }) => ({
  display: 'flex',
  flexWrap: 'wrap',
  width: '100%',
  gap: '1rem',
  [theme.breakpoints.up('md')]: {
    gap: 0,
  },
}));

const LiveStats = () => {
  const { stats } = useLiveStats();
  return (
    <Root>
      {stats.map((item) => (
        <StatsItem
          key={item.title}
          title={item.title}
          value={item.value}
          hint={item.hint}
          progress={item.progress}
        />
      ))}
    </Root>
  );
};

export default LiveStats;
