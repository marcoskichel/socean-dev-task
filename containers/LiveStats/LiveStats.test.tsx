import { ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';
import { useLiveStats } from '../../providers/LiveStats/LiveStatsProvider';
import theme from '../../styles/theme';
import LiveStats from './LiveStats';

jest.mock('../../providers/LiveStats/LiveStatsProvider');

const mockUseLiveStats = useLiveStats as jest.Mock;

mockUseLiveStats.mockReturnValue({
  stats: [
    {
      title: 'Total Value Locked',
      value: '$61.70M USD',
      hint: 'SOL/USD: $87.9800',
    },
    {
      title: 'SOL/scnSOL in Pool',
      value: '701.31K SOL',
      hint: '684,440 scnSOL',
    },
    {
      title: '7D APY',
      value: '6.08%',
      hint: '12H APY: 7.14%',
    },
    {
      title: 'SOL/scnSOL Price',
      value: '1.0247 SOL',
      hint: '= 1 scnSOL',
    },
    {
      title: 'EPOCH 272',
      value: '19%',
      hint: 'ETA: 2d 5h 16m 10s',
      progress: 19,
    },
  ],
});

describe('LiveStats', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <LiveStats />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
