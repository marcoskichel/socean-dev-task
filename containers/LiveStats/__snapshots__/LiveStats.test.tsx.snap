// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveStats renders 1`] = `
<DocumentFragment>
  <div
    class="css-fgs0i6"
  >
    <div
      class="css-fi8v5m"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1je6kat-MuiTypography-root"
      >
        Total Value Locked
      </h6>
      <div
        class="css-1rhchyf"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-hynxzr-MuiTypography-root"
      >
        $61.70M USD
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-1vuxh5z-MuiTypography-root"
      >
        SOL/USD: $87.9800
      </span>
    </div>
    <div
      class="css-fi8v5m"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1je6kat-MuiTypography-root"
      >
        SOL/scnSOL in Pool
      </h6>
      <div
        class="css-1rhchyf"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-hynxzr-MuiTypography-root"
      >
        701.31K SOL
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-1vuxh5z-MuiTypography-root"
      >
        684,440 scnSOL
      </span>
    </div>
    <div
      class="css-fi8v5m"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1je6kat-MuiTypography-root"
      >
        7D APY
      </h6>
      <div
        class="css-1rhchyf"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-hynxzr-MuiTypography-root"
      >
        6.08%
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-1vuxh5z-MuiTypography-root"
      >
        12H APY: 7.14%
      </span>
    </div>
    <div
      class="css-fi8v5m"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1je6kat-MuiTypography-root"
      >
        SOL/scnSOL Price
      </h6>
      <div
        class="css-1rhchyf"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-hynxzr-MuiTypography-root"
      >
        1.0247 SOL
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-1vuxh5z-MuiTypography-root"
      >
        = 1 scnSOL
      </span>
    </div>
    <div
      class="css-fi8v5m"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1je6kat-MuiTypography-root"
      >
        EPOCH 272
      </h6>
      <div
        class="css-1rhchyf"
      >
        <span
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="19"
          class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate css-5js6s3-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate css-8m0zll-MuiLinearProgress-bar1"
            style="transform: translateX(-81%);"
          />
        </span>
      </div>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-hynxzr-MuiTypography-root"
      >
        19%
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-1vuxh5z-MuiTypography-root"
      >
        ETA: 2d 5h 16m 10s
      </span>
    </div>
  </div>
</DocumentFragment>
`;
